core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1800,6 +1800,15 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  lazy val isInInterpreter: Boolean = {
+    try {
+      val interpClass = classForName("spark.repl.Main")
+      interpClass.getMethod("interp").invoke(null) != null
+    } catch {
+      case _: ClassNotFoundException => false
```

Comment:
@andrewor14 @squito please, note the change to the return value I introduced on this line guys.
This is controversial to master (https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala#L242) where we return `true` in case of getting an exception. Still trying to understand why it was done that way. If this was on purpose and had a reason behind, I'd be grateful for some piece of explanation.

Since this val now resides in Utils and is therefore presented to the public, I am convinced there need to be no confusion around its logic.
Hope it makes sense what I'm saying.